ui/react-components/utils/AppointmentListSidePanelWrapper/getFilteredNodesOnSearch.js

```diff
@@ -0,0 +1,43 @@
+const getFilteredNodesOnSearch = (transformedTreeData, searchText) => {
+  const searchFilteredNodesList = transformedTreeData
+    .filter(element => {
+      return (
+        element.label
+          .trim()
+          .toLowerCase()
+          .includes(searchText.toLowerCase()) ||
+        (element.children &&
+          element.children.some(subElement =>
+            subElement.label
+              .trim()
+              .toLowerCase()
+              .includes(searchText.toLowerCase())
+          ))
+      );
+    })
+    .map(ele => {
+      const filteredList = Object.assign(
```

Comment:
Similar here, something like `getMachingElementsOrChildren`.

Reply:
Now we have refactor that function